### PR TITLE
azure - marked for op - time zone reset

### PR DESCRIPTION
**Pull request by jomalsan**
When filtering a list of resources, the same instance of `TagActionFilter` gets used. So when one resource in the list has a timezone, `self.current_date` becomes timezone aware for all following resources. If it then process a resource with no timezone information we need to set `self.current_date` to be timezone naive again to avoid the error reported in #4671

### CODE REVIEW

File: tools/c7n_azure/c7n_azure/filters.py [hunk 1]
```diff
@@ -354,6 +353,9 @@ def __call__(self, i):
             # if action_date is timezone aware, set to timezone provided
             action_date = action_date.astimezone(self.tz)
             self.current_date = now(tz=self.tz)
+        else:
+            # if it is not need to reset to timezone naive
```
Comment (aluong) [inline tools/c7n_azure/c7n_azure/filters.py]: (minor) can you update the comment to be a little clearer.

Comment (logachev) [inline tools/c7n_azure/c7n_azure/filters.py]: Good catch! I think we can get rid of `self.current_date` entirely and replace it with function-scoped `current_date`

Comment (jomalsan) [reply tools/c7n_azure/c7n_azure/filters.py]: Good idea, that should really clarify things

**Comment by logachev** [comment]
opened https://github.com/cloud-custodian/cloud-custodian/issues/4818 because same issue exists for AWS

**Comment by jomalsan** [comment]
Will add new unit tests to increase the code coverage. Goal is to cover the entire `__call__` method

**Pull request merged**